go/integTest/shared_commands_test.go

```diff
@@ -0,0 +1,172 @@
+// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+package integTest
+
+import (
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/valkey-io/valkey-glide/go/glide/api"
+)
+
+const (
+	keyName      = "key"
+	initialValue = "value"
+	anotherValue = "value2"
+)
+
+func (suite *GlideTestSuite) TestSetAndGet_noOptions() {
+	suite.runWithDefaultClients(func(client api.BaseClient) {
+		suite.verifyOK(client.Set(keyName, initialValue))
+		result, err := client.Get(keyName)
+
+		assert.Nil(suite.T(), err)
+		assert.Equal(suite.T(), initialValue, result)
+	})
+}
+
+func (suite *GlideTestSuite) TestSetAndGet_byteString() {
+	suite.runWithDefaultClients(func(client api.BaseClient) {
+		invalidUTF8Value := "\xff\xfe\xfd"
+		suite.verifyOK(client.Set(keyName, invalidUTF8Value))
+		result, err := client.Get(keyName)
+
+		assert.Nil(suite.T(), err)
+		assert.Equal(suite.T(), invalidUTF8Value, result)
+	})
+}
+
+func (suite *GlideTestSuite) TestSetWithOptions_ReturnOldValue() {
+	suite.runWithDefaultClients(func(client api.BaseClient) {
+		suite.verifyOK(client.Set(keyName, initialValue))
+
+		opts := &api.SetOptions{ReturnOldValue: true}
+		result, err := client.SetWithOptions(keyName, anotherValue, opts)
+
+		assert.Nil(suite.T(), err)
+		assert.Equal(suite.T(), initialValue, result)
+	})
+}
+
+func (suite *GlideTestSuite) TestSetWithOptions_OnlyIfExists_overwrite() {
+	suite.runWithDefaultClients(func(client api.BaseClient) {
+		key := "TestSetWithOptions_OnlyIfExists_overwrite"
```

Comment:
Thanks for giving different readable names - it helps with debugging

Reply:
Acknowledged :)